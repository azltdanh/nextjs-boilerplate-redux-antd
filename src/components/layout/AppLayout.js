import Head from 'next/head';
import { Layout, Row, Col } from 'antd';
import { APP_TITLE } from '@constants';
import AppLogo from './AppLogo';
import AppMenu from './AppMenu';
import AppMenuUser from './AppMenuUser';

import '@assets/main.less';

const AppLayout = ({ children, title = '' }) => {
  const pageTitle = `${APP_TITLE}${title ? ` - ${title}` : ''}`;
  return (
    <Layout className="app-layout">
      <Head>
        <link rel="icon" href="/favicon.ico" />
        <meta
          name="description"
          content="VIBE - A great start to every trip!"
        />
        <meta name="og:title" content={pageTitle} />
        <meta name="twitter:card" content="summary_large_image" />
        <title>{pageTitle}</title>
      </Head>
      <Layout.Header>
        <Row className="app-header" justify="space-between">
          <Col>
            <Row>
              <Col>
                <AppLogo />
              </Col>
              <Col>
                <AppMenu />
              </Col>
            </Row>
          </Col>
          <Col>
            <AppMenuUser />
          </Col>
        </Row>
      </Layout.Header>
      <Layout.Content>
        <main className="app-content">{children}</main>
      </Layout.Content>
      <Layout.Footer className="app-footer">
        VIBE. All rights reserved. Â© Copyright 2020
      </Layout.Footer>
    </Layout>
  );
};

export default AppLayout;
