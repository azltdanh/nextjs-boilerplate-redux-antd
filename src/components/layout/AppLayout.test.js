import AppLayout from './AppLayout';

describe(AppLayout.name, () => {
  it('renders', () => {
    const wrapper = shallow(<AppLayout />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with props', () => {
    const wrapper = shallow(<AppLayout title="Title" />);
    expect(wrapper).toMatchSnapshot();
  });
});
