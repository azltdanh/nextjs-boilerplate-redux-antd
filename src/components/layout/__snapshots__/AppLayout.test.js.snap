// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout renders 1`] = `
<Layout
  className="app-layout"
>
  <Head>
    <link
      href="/favicon.ico"
      rel="icon"
    />
    <meta
      content="VIBE - A great start to every trip!"
      name="description"
    />
    <meta
      content="VIBE"
      name="og:title"
    />
    <meta
      content="summary_large_image"
      name="twitter:card"
    />
    <title>
      VIBE
    </title>
  </Head>
  <Header>
    <Row
      className="app-header"
      justify="space-between"
    >
      <Col>
        <Row>
          <Col>
            <AppLogo />
          </Col>
          <Col>
            <AppMenu />
          </Col>
        </Row>
      </Col>
      <Col>
        <AppMenuUser />
      </Col>
    </Row>
  </Header>
  <Content>
    <main
      className="app-content"
    />
  </Content>
  <Footer
    className="app-footer"
  >
    VIBE. All rights reserved. © Copyright 2020
  </Footer>
</Layout>
`;

exports[`AppLayout renders with props 1`] = `
<Layout
  className="app-layout"
>
  <Head>
    <link
      href="/favicon.ico"
      rel="icon"
    />
    <meta
      content="VIBE - A great start to every trip!"
      name="description"
    />
    <meta
      content="VIBE - Title"
      name="og:title"
    />
    <meta
      content="summary_large_image"
      name="twitter:card"
    />
    <title>
      VIBE - Title
    </title>
  </Head>
  <Header>
    <Row
      className="app-header"
      justify="space-between"
    >
      <Col>
        <Row>
          <Col>
            <AppLogo />
          </Col>
          <Col>
            <AppMenu />
          </Col>
        </Row>
      </Col>
      <Col>
        <AppMenuUser />
      </Col>
    </Row>
  </Header>
  <Content>
    <main
      className="app-content"
    />
  </Content>
  <Footer
    className="app-footer"
  >
    VIBE. All rights reserved. © Copyright 2020
  </Footer>
</Layout>
`;
